protocol/messenger.go

```diff
@@ -1428,6 +1428,13 @@ func (m *Messenger) handleEncryptionLayerSubscriptions(subscriptions *encryption
 					m.logger.Error("failed to clean processed messages", zap.Error(err))
 				}
 
+			case keys := <-subscriptions.NewHashRatchetKeys:
+				if m.communitiesManager == nil {
+					continue
+				}
+				if err := m.communitiesManager.NewHashRatchetKeys(keys); err != nil {
+					m.logger.Error("failed to invalidate cache for decrypted communities", zap.Error(err))
+				}
```

Comment:
Shouldn't we also try to decrypt the description here and update the client with newly decrypted data in community?

Reply:
That's taken care separately, as part of the retrieve messages loop

Reply:
What if keys are handled after the description? :thinking: Wouldn't we need to wait for next description or restart the app to properly decrypt missing data?

Reply:
that's a good point, let me check

Reply:
ok, so I have added invalidating the cache before processing communities messages ,that should take care of the race condition, it's a bit redundant, as we might invalidate the cache multiple times, but it's a bit safer as it catches other cases where we might receive keys, well spotted!